src/etl/Batch_Ingestion_Script.py

```diff
@@ -1,81 +1,213 @@
+
+    Returns:
+        tuple: A tuple containing:
+            - found (bool): Whether at least one pollutant value has been found.
+            - datetime (str): The ISO_8601 UTC datetime indicating the time the data refers to.
+            - index_name (str): The name of the returned water quality index.
+            - qualification (str): A textual representation of the water quality (e.g., "Good", "Moderate").
+            - description (str): A more detailed textual assessment of the water quality.
+            - color (str): A hexadecimal color code representing the water quality level.
+            - health_recommendations (str): General health recommendations for the population, if available.
+            - response (str): The original JSON response as a string for reference or future processing.
+    
+    Example:
+        response = '{"found": true, "datetime": "2024-09-10T05:00:00.000Z", "index": {...}, "health_recommendations": {...}}'
+        found, datetime, index_name, qualification, description, color, health_recommendations, original_response = extract_data_from_response(response)
+    """
+    # Parse the JSON response
+    data = json.loads(water_quality_data)
+
+    # Extract critical information
+    found = data.get('found', False)
+    datetime = data.get('datetime', None)
+
+    index = data.get('index', {})
+    index_name = index.get('index_name', None)
+    qualification = index.get('qualification', None)
+    description = index.get('description', None)
+    color = index.get('color', None)
+
+    # Health recommendations (optional)
+    health_recommendations = data.get('health_recommendations', {}).get('all', None)
+
+    # Return extracted information and original JSON
+    return found, datetime, index_name, qualification, description, color, health_recommendations, response
 
-#fetch data using mapPartitions
-water_quality_rdd = cities_df.rdd.mapPartitions(fetch_data_for_partition).toDF()
-first_element = water_quality_rdd.take(1)
-print(first_element)
 
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC #### Obtain city coordinates
+
+# COMMAND ----------
 
-#water_quality_df = spark.createDataFrame(water_quality_rdd, schema)
 
-#show df
-#water_quality_df.show()
+#set cities
+cities = ["Paris", "Nice", "Bordeaux"]
+country="France"
+
+#configure geocoder to fetch coordinates from Azure Maps API
+maps_key = "your-azure-maps-key"
+geocoder = AzureMaps(subscription_key=maps_key)
+
+#store output in dict if they exist, and store coords as tuples
+city_geocodes = {}
+for city in cities:
+    lat, lng = fetch_geocodes(city, country)
+    if lat is not None and lng is not None:
+        key = f"{city}, {country}"
+        city_geocodes[key] = (lat, lng)
+    else:
+        print(f"Location coordinates for {city}, {country} not found")
+
+print(city_geocodes)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC #### Obtain API Data and Create Spark DF
+
+# COMMAND ----------
+
+    
+meersens_api_key = "your-meersens-key"
```

Comment:
where is this being fetched from? Its just stated as a string for now? Is this a dummy api key?